pkg/util/subnet/subnet.go

```diff
@@ -13,25 +13,43 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/apparentlymart/go-cidr/cidr"
+	maoclient "github.com/openshift/machine-api-operator/pkg/generated/clientset/versioned"
 
 	"github.com/Azure/ARO-RP/pkg/api"
-	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
 	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
 )
 
+type Subnet struct {
+	ResourceID string
+	IsMaster   bool
+}
+
 type Manager interface {
 	Get(ctx context.Context, subnetID string) (*mgmtnetwork.Subnet, error)
 	GetHighestFreeIP(ctx context.Context, subnetID string) (string, error)
 	CreateOrUpdate(ctx context.Context, subnetID string, subnet *mgmtnetwork.Subnet) error
+
+	// ListFromCluster uses kube clients, instead of azure go get cluster
+	// subnets
+	ListFromCluster(ctx context.Context) ([]Subnet, error)
```

Comment:
Somehow this doesn't seem like a good fit: now this interface both deals with Azure API and with clsuters... Responsibility seem to be too broad. Maybe we need a separate thing for interacting with clusters?

Also I don't like the fact that in many cases we are going to call `NewManager` with `nil` as a first argument which is going to be confusing: you have to know when to provide it and when not to provide it.

What do you think?

Reply:
I dont like it too, but we need it in a re-usable way. So open for suggestions. Separate interface? Separate package?

Reply:
@mjudeikis Yes, I was think about a separate interface & struct for it. I think same package is ok.